### PR TITLE
Kevin/group joining endpoint

### PR DESCRIPTION
**Pull request by KevinetPeng**
Fixed groupcreate bugs, user login bugs, and implemented group joining.

Note: for some reason, MongoDB isn't throwing an error if the groupCode or spotifyID doesn't match an existing object in our database. This means that the API isn't responding with an error when it should. This won't actually affect our usage of the group-create or group-joining if we provide the correct request body though.

### CODE REVIEW

File: backend/router.js [hunk 1]
```diff
@@ -6,9 +6,12 @@ const Controller = require("./Controller.js");
 router.get("/token/:token/user", Controller.getUser)
 
 //define post request for making a new group
-router.post("/group/create", Controller.createGroup);
+router.post("/groups/create", Controller.createGroup);
```
Comment (brendanlai) [inline backend/router.js:9]: personally prefer the group non-plural route name as it is more indicative (single group being created not multiple). but this is a matter of opinion

File: backend/router.js [hunk 2]
```diff
@@ -6,9 +6,12 @@ const Controller = require("./Controller.js");
...
 
 //route post request for new user login
 router.post("/users/login", Controller.loginUser);
 
+//define post request for joining a group
+router.post("/groups/join", Controller.joinGroup);
```
Comment (brendanlai) [inline backend/router.js:15]: same as the above comment

File: backend/Controller.js [hunk 3]
```diff
@@ -1,128 +1,173 @@
...
+  //add userID to the group object
+  Group.updateOne(
+    { groupCode: req.body.groupCode }, //filter
+    { $addToSet: { users: req.body.spotifyID } }
+  )
+    .then(() => {
+      //add groupCode to user object
+      User.updateOne(
+        { userID: req.body.spotifyID }, //filter
+        { $addToSet: { groups: req.body.groupCode } }
```
Comment (brendanlai) [inline backend/Controller.js:150]: NOTE: I can join a group I already exist in. Should probably introduce a check that disables users from joining a group they are already a part of

**Review by Brendanlai — COMMENTED** [review]
Everything looks good to me. Note that there is a bug for a user joining the same group multiple times but if this is difficult can be pushed back until further but keep note.
Some other minor things about the syntax of the routes but it's not that important

**Comment by Brendanlai** [comment]
> Everything looks good to me. Note that there is a bug for a user joining the same group multiple times but if this is difficult can be pushed back until further but keep note.
> Some other minor things about the syntax of the routes but it's not that important

Sike, its just the response that needs to be fixed. the functional adding of a member is one time joining

**Comment by KevinetPeng** [comment]
Yup, it seems to be an issue with how MongoDB decides to throw errors. For example, if the filter of a updateOne method can't find a matching database object, an error isn't thrown, and its the same case here with the operation for addToSet.

**Pull request merged**